src/charm.py

```diff
@@ -82,7 +82,7 @@ def __init__(self, *args):
 
         self.framework.observe(self.on.agent_pebble_ready, self.on_pebble_ready)
         self.framework.observe(
-            self.on[REMOTE_WRITE_RELATION_NAME].relation_changed, self.on_remote_write_changed
+            self._remote_write.on.endpoints_changed, self.on_remote_write_changed
```

Comment:
Isn’t it likely we have the _same_ bug on line 88? And a similar one on 92/95?

Also: where are the tests? ;-)

Reply:
Fixed in 621103cb14fa759dce2ee87bca28e1274241c3ae